### PR TITLE
[Feature] vhash join profile collect

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/join/vacquire_list.hpp [hunk 1]
```diff
@@ -25,7 +25,10 @@ namespace doris::vectorized {
 template <typename Element, int batch_size = 8>
 struct AcquireList {
     using Batch = Element[batch_size];
+
+public:
```
Comment (HappenLee) [inline be/src/vec/exec/join/vacquire_list.hpp:29]: struct 默认就是public，这里可以不用加

File: be/src/vec/exec/join/vhash_join_node.cpp [hunk 2]
```diff
@@ -549,6 +553,11 @@ Status HashJoinNode::close(RuntimeState* state) {
     }
 
     if (_vother_join_conjunct_ptr) (*_vother_join_conjunct_ptr)->close(state);
+
+    _mem_tracker->Release(_hash_table_bytes);
```
Comment (HappenLee) [inline be/src/vec/exec/join/vhash_join_node.cpp:580]: 这个代码重构一下

File: be/src/vec/exec/join/vhash_join_node.cpp [hunk 3]
```diff
@@ -588,7 +600,7 @@ Status HashJoinNode::get_next(RuntimeState* state, Block* output_block, bool* eo
                             auto &null_map_val = _null_map_column->get_data();
                             return extract_probe_join_column(
                                     _probe_block, null_map_val, _probe_columns,
-                                    _probe_ignore_null);
+                                    _probe_ignore_null,*_probe_expr_call_timer);
```
Comment (HappenLee) [inline be/src/vec/exec/join/vhash_join_node.cpp]: 逗号这里补充一个空格

File: be/src/vec/exec/join/vhash_join_node.cpp [hunk 4]
```diff
@@ -694,19 +706,26 @@ Status HashJoinNode::_hash_table_build(RuntimeState* state) {
...
+        
         RETURN_IF_ERROR(_process_build_block(block));
+        RETURN_IF_LIMIT_EXCEEDED(state, "Hash join, while constructing the hash table.");
     }
     return Status::OK();
 }
 
 Status HashJoinNode::extract_build_join_column(Block& block, NullMap& null_map,
-                                            ColumnRawPtrs& raw_ptrs, bool& ignore_null) {
+                                            ColumnRawPtrs& raw_ptrs, bool& ignore_null,RuntimeProfile::Counter& expr_call_timer) {
```
Comment (HappenLee) [inline be/src/vec/exec/join/vhash_join_node.cpp]: 逗号空格

File: be/src/vec/exec/join/vhash_join_node.cpp [hunk 5]
```diff
@@ -731,12 +750,14 @@ Status HashJoinNode::extract_build_join_column(Block& block, NullMap& null_map,
 }
 
 Status HashJoinNode::extract_probe_join_column(Block& block, NullMap& null_map,
-                                            ColumnRawPtrs& raw_ptrs, bool& ignore_null) {
+                                            ColumnRawPtrs& raw_ptrs, bool& ignore_null,RuntimeProfile::Counter& expr_call_timer) {
```
Comment (HappenLee) [inline be/src/vec/exec/join/vhash_join_node.cpp]: 同上

File: be/src/vec/exec/join/vhash_join_node.cpp [hunk 6]
```diff
@@ -787,7 +810,7 @@ Status HashJoinNode::_process_build_block(Block& block) {
             [&](auto&& arg) -> Status {
                 using HashTableCtxType = std::decay_t<decltype(arg)>;
                 if constexpr (!std::is_same_v<HashTableCtxType, std::monostate>) {
-                    return extract_build_join_column(acquired_block, null_map_val, raw_ptrs, has_null);
+                    return extract_build_join_column(acquired_block, null_map_val, raw_ptrs, has_null,*_build_expr_call_timer);
```
Comment (HappenLee) [inline be/src/vec/exec/join/vhash_join_node.cpp]: 同上

File: be/src/vec/exec/join/vhash_join_node.h [hunk 7]
```diff
@@ -193,24 +184,24 @@ class HashJoinNode : public ::doris::ExecNode {
     const bool _match_all_probe; // output all rows coming from the probe input. Full/Left Join
     const bool _match_one_build; // match at most one build row to each probe row. Left semi Join
     const bool _match_all_build; // output all rows coming from the build input. Full/Right Join
-    bool _build_unique;    // build a hash table without duplicated rows. Left semi/anti Join
+    bool _build_unique;          // build a hash table without duplicated rows. Left semi/anti Join
```
Comment (HappenLee) [inline be/src/vec/exec/join/vhash_join_node.h:187]: 这个空格是？

**Review by HappenLee — COMMENTED** [review]
有些小问题，请修改一下

File: be/src/vec/exec/join/vhash_join_node.cpp (hunk 2 above)
Comment (BiteTheDDDDt) [reply be/src/vec/exec/join/vhash_join_node.cpp]: done.

**Pull request merged**